src/gui/vsDevice.cpp

```diff
@@ -296,6 +315,25 @@ void VsDevice::reconcileAgentConfig(QJsonDocument newState)
     }
 }
 
+// initPinger intializes the pinger used to generate network latency statistics for
+// Virtual Studio
+VsPinger* VsDevice::startPinger(VsServerInfo* studioInfo)
+{
+    QString id        = studioInfo->id();
+    QString sessionId = studioInfo->sessionId();
+    QString host      = studioInfo->sessionId();
+    host.append(QString::fromStdString(".jacktrip.cloud"));
+
+    m_pinger = new VsPinger(QString::fromStdString("wss"), host,
+                            QString::fromStdString("/ping"));
+}
```

Comment:
Probably noob language syntax question here, but do we need a `return m_pinger` here to match the method signature? I know some language automatically return the last time but I'm not quite sure if that's true in this case

Reply:
hmm interesting question. I don't think it threw any compiler warnings or errors and I'm not actually sure why. Maybe because the function signature returns a pointer it fills in nullptr if there's no return statement?

Reply:
It generally won't throw an error unless you turn additional warnings on, but it is usually undefined behaviour. It probably won't return null, but instead the last thing that was in the relevant register. Which is definitely not a good thing for a pointer.

Reply:
(What IDE are you using? There are usually settings in most that will allow the syntax analyser to warn you about these sorts of things as you code.)

Reply:
I'm using VSCode, though I haven't yet really figured out the C/C++ extensions and settings that would help with these sorts of things